core/capabilities/ccip/ccip_integration_tests/usdcreader/usdcreader_test.go

```diff
@@ -201,7 +207,147 @@ func Test_USDCReader_MessageHashes(t *testing.T) {
 		})
 	}
 }
+func Benchmark_MessageHashes(b *testing.B) {
+	finalityDepth := 5
+
+	// Adding a new parameter: tokenCount
+	testCases := []struct {
+		name       string
+		msgCount   int
+		startNonce uint64
+		tokenCount int
+	}{
+		{"Small_Dataset", 100, 1, 5},
+		{"Medium_Dataset", 10_000, 1, 10},
+		{"Large_Dataset", 100_000, 1, 50},
```

Comment:
Nice, was there any major difference between these 3 datasets?

Reply:
```
// Benchmark_MessageHashes/Small_Dataset-14       3723        272421 ns/op       126949 B/op      2508 allocs/op
// Benchmark_MessageHashes/Medium_Dataset-14       196       6164706 ns/op      1501435 B/op     20274 allocs/op
// Benchmark_MessageHashes/Large_Dataset-14          7     163930268 ns/op     37193160 B/op    463954 allocs/op
```

It's going linearly. I think that the remappings we have (first the messageSentEvents and then out) are a big offender (apart for the GetQuery).